src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -1386,4 +1392,91 @@ private boolean checkIfUserIsMemberOfCourseOrganizations(User user, Course cours
         }
         return isMember;
     }
+
+    /**
+     * GET /courses/{courseId}/management-detail : Gets the data needed for the course management detail view
+     *
+     * @param courseId the id of the course
+     * @return the ResponseEntity with status 200 (OK) and the body, or with status 404 (Not Found)
+     */
+    @GetMapping("/courses/{courseId}/management-detail")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<CourseManagementDetailViewDTO> getCourseDTOForDetailView(@PathVariable Long courseId) {
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.TEACHING_ASSISTANT, course, null);
+
+        var exercises = exerciseRepository.findAllExercisesByCourseId(courseId);
+        var includedExercises = exercises.stream().filter(Exercise::isCourseExercise)
+                .filter(exercise -> !exercise.getIncludedInOverallScore().equals(IncludedInOverallScore.NOT_INCLUDED)).collect(Collectors.toSet());
+        var averageScoreForCourse = participantScoreRepository.findAvgScore(includedExercises);
+        averageScoreForCourse = averageScoreForCourse != null ? averageScoreForCourse : 0.0;
+        var reachablePoints = includedExercises.stream().map(Exercise::getMaxPoints).collect(Collectors.toSet()).stream().mapToDouble(Double::doubleValue).sum();
```

Comment:
You use a lot of var here. At least I'd prefer real types here.

Reply:
We could have a little discussion about that in the next weekly meeting. I personally started using var a lot since working on this project because it is used very heavily here. If we decide to use less vars i can refactor this, of course

Reply:
Added it to the meeting discussion for next meeting

Reply:
Sounds good